lib/honeybadger/notice.rb

```diff
@@ -200,6 +205,7 @@ def as_json(*args)
       {
         api_key: s(api_key),
         notifier: NOTIFIER,
+        breadcrumbs: s(breadcrumbs.to_h),
```

Comment:
Ah, I see we're also sanitizing the breadcrumbs here--so maybe I'm misunderstanding the purpose of `Cleaner`? Why do we need two types of cleaning?

Reply:
Just to be clear on this one, I think it should still happen here, even if we have a stricter cleaner class that runs before it. Breadcrumb data can change between `Honeybadger.notify` and sending the notice through `before_notify` callbacks.

One thing that may be a bit incongruous based on my [above comment](https://github.com/honeybadger-io/honeybadger-ruby/pull/312#discussion_r291401232) is that the result of cleaning any data added at send-time via `before_notify` (via `s()` will not be the same as cleaning the data beforehand (e.g., the `Cleaner` is used for data inputed with `Honeybadger.add_breadcrumb`, but the `Sanitizer` if add with a `before_notify` callback).